packages/x-charts-pro/src/internals/material/components/BasePopper.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import Popper, { PopperProps } from '@mui/material/Popper';
+import MUIFocusTrap from '@mui/material/Unstable_TrapFocus';
+import ClickAwayListener from '@mui/material/ClickAwayListener';
+import Grow from '@mui/material/Grow';
+import Paper from '@mui/material/Paper';
+import { ChartBasePopperProps } from '../../slots/chartBaseSlotProps';
+
+function clickAwayWrapper(props: ChartBasePopperProps, content: any) {
+  if (props.onClickAway === undefined) {
+    return content;
+  }
+  return (
+    <ClickAwayListener
+      onClickAway={props.onClickAway as any}
+      touchEvent={props.clickAwayTouchEvent}
+      mouseEvent={props.clickAwayMouseEvent}
+    >
+      {content}
+    </ClickAwayListener>
+  );
+}
+
+function focusTrapWrapper(props: ChartBasePopperProps, content: any) {
+  if (props.focusTrap === undefined) {
+    return content;
+  }
+  return (
+    <MUIFocusTrap open disableEnforceFocus disableAutoFocus>
+      <div tabIndex={-1}>{content}</div>
+    </MUIFocusTrap>
+  );
+}
+
+function wrappers(props: ChartBasePopperProps, content: any) {
+  return focusTrapWrapper(props, clickAwayWrapper(props, content));
+}
+
+const transformOrigin = {
+  'bottom-start': 'top left',
+  'bottom-end': 'top right',
+};
+
+export function BasePopper(props: ChartBasePopperProps) {
```

Comment:
Basically a copy of Data Grid's `BasePopper` ([source](https://github.com/mui/mui-x/blob/8fc7c02ec4031dd606791cf4c52f7d1e0896b371/packages/x-data-grid/src/material/index.tsx#L491))